env_framework/track.py

```diff
@@ -0,0 +1,57 @@
+"""
+This is the characteristics of the track. This file acts as the main physics engine.
+
+A track field consists of equally sized tiles, with upper left as (0, 0), upper right as (0, 1999),
+lower left corner as (499, 0), and lower right corner as (499, 1999)
+"""
+
+import numpy as np
+from car import CarState
+from model import Action
+
+FieldTile = np.dtype(
+    [("type_id", np.uint8), ("is_center", np.bool_), ("distance", np.float64)]
+)
+
+
+class TrackField:
+    __slots__ = ["field"]
+    field: np.ndarray
+
+    def __init__(self) -> None:
+        self.field = np.zeros((500, 2000), dtype=FieldTile)
+
+
+class CarView:
+    __slots__ = ["upper_left", "field"]
+    upper_left: tuple[int, int]
+    field: np.ndarray
+
+    def __init__(self, field: TrackField, x: int, y: int, view_distance: int) -> None:
+        """
+        Creates a new CarView
+
+        Parameters:
+            field: the TrackeField of the car
+            x: x coordinate of the car
+            y: y coordinate of the car
+            view_distance: distance that car can see in one direction from the center.
+        """
+        self.upper_left = (x, y)
```

Comment:
field is right if (x,y) is car center's tile coordinates. 

In that case 
self.upper_left = (x - view_distance, y - view_distance)